src/Styleguide/Utils/Responsive.tsx

```diff
@@ -40,6 +47,14 @@ export class ResponsiveProvider extends React.Component<
     }
   }
 
+  shouldComponentUpdate(nextProps, nextState) {
+    if (nextProps.children !== this.props.children) return true
+    if (shallowEqual(this.state.breakpoints, nextState.breakpoints)) {
+      return false
+    }
+    return true
+  }
```

Comment:
This requires some testing (especially because i'm not sure about shallow comparisons of  `children` as a data structure), but maybe `React.PureComponent` would work well here.